sdk/inc/azure/core/az_platform.h

```diff
@@ -53,6 +87,70 @@ AZ_NODISCARD az_result az_platform_clock_msec(int64_t* out_clock_msec);
  */
 AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds);
 
+/**
+ * @brief Called on critical error. This function should not return.
+ *
+ * @details In general, this function should cause the device to reboot or the main process to
+ *          crash or exit.
+ */
+void az_platform_critical_error();
+
+/**
+ * @brief Gets a positive pseudo-random integer.
+ * 
+ * @param[out] out_random A pseudo-random number greater than 0.
+ *  *
+ * @retval #AZ_OK Success.
+ * @retval #AZ_ERROR_DEPENDENCY_NOT_PROVIDED No platform implementation was supplied to support this
+ * function.
+ * 
+ * @note This is NOT cryptographically secure.
+ */
+AZ_NODISCARD az_result az_platform_get_random(int32_t* out_random);
+
+
+/**
+ * @brief Create a timer object.
+ *
+ * @param[in] callback SDK callback to call when timer elapses.
+ * @param[in] sdk_data SDK data associated with the timer.
+ * @param[out] out_timer_handle The timer handle.
```

Comment:
swap timer to top and make correct name:
```suggestion
 * @param[out] timer The timer handle.
 * @param[in] callback SDK callback to call when timer elapses.
 * @param[in] sdk_data SDK data associated with the timer.
```